Compilation error when creating an AutoHashMap with an Enum key

Hi !

**Zig version**:  _0.4.0_

**Case**: I want to build an new type that is an AutoHashMap with an Enum as the Key type.

**Resulting behavior**: I encounter a compilation error because most of calls to the `autoHash` function miss a 3rd parameter.

Here is a minimal test case:

```zig
const AutoHashMap = @import("std").AutoHashMap;

pub const Animal = enum {
    Cat,
    Dog,
    Fish
};

pub fn main() void {
    const AnimalSet = AutoHashMap(Animal, u32);
}
```


```bash
bug.zig:10:34: error: expected 2 arguments, found 1
    const AnimalSet = AutoHashMap(Animal);
                                 ^
/usr/lib/zig/std/hash_map.zig:13:5: note: declared here
pub fn AutoHashMap(comptime K: type, comptime V: type) type {
    ^
# zig run src/enum_auto_hashmap_bug.zig
/usr/lib/zig/std/hash_map.zig:489:47: error: expected 3 arguments, found 2
        builtin.TypeId.Enum => return autoHash(@enumToInt(key), rng),
                                              ^
/usr/lib/zig/std/hash_map.zig:450:28: note: called from here
            return autoHash(key, &rng.random, u32);
                           ^
/usr/lib/zig/std/hash_map.zig:464:5: note: declared here
pub fn autoHash(key: var, comptime rng: *std.rand.Random, comptime HashInt: type) HashInt {
    ^
```

--- a/std/hash_map.zig
+++ b/std/hash_map.zig
@@ -558,12 +558,12 @@ pub fn autoHash(key: var, comptime rng: *std.rand.Random, comptime HashInt: type
         },
 
         builtin.TypeId.Float => |info| {
-            return autoHash(@bitCast(@IntType(false, info.bits), key), rng);
+            return autoHash(@bitCast(@IntType(false, info.bits), key), rng, HashInt);
         },
-        builtin.TypeId.Bool => return autoHash(@boolToInt(key), rng),
-        builtin.TypeId.Enum => return autoHash(@enumToInt(key), rng),
-        builtin.TypeId.ErrorSet => return autoHash(@errorToInt(key), rng),
-        builtin.TypeId.Promise, builtin.TypeId.Fn => return autoHash(@ptrToInt(key), rng),
+        builtin.TypeId.Bool => return autoHash(@boolToInt(key), rng, HashInt),
+        builtin.TypeId.Enum => return autoHash(@enumToInt(key), rng, HashInt),
+        builtin.TypeId.ErrorSet => return autoHash(@errorToInt(key), rng, HashInt),
+        builtin.TypeId.Promise, builtin.TypeId.Fn => return autoHash(@ptrToInt(key), rng, HashInt),
 
         builtin.TypeId.BoundFn,
         builtin.TypeId.ComptimeFloat,

